testing/scenario_app/android/README.md

```diff
@@ -1,102 +1,364 @@
+
+## Introduction
+
+This package simulates a Flutter app that uses the engine (`dart:ui`) only, in
+conjunction with Android-specific embedding code that simulates the use of the
+engine in a real app (such as plugins and platform views).
+
+A custom test runner, [`run_android_tests.dart`](bin/run_android_tests.dart), is
+used to run the tests on a connected device or emulator, and report the results,
+including screenshots (golden files) and logs from `adb logcat`.
+
+In the following architecture diagram:
+
+> <a
+>   href="https://github.com/flutter/flutter/wiki/The-Engine-architecture"
+>   title="The Engine architecture"> <img
+>   width="300"
+>   alt="Anatomy of a Flutter app"
+>   src="https://raw.githubusercontent.com/flutter/engine/main/docs/app_anatomy.svg"
+> /> </a>
+
+- The Dart app is represented by [`lib/main.dart`](../lib/main.dart).
+- There is no framework code.
+- `dart:ui` and the engine are the same as in a real app.
+- Android-specific application code is in this directory
+  ([`android/`](./)).
+- The runner is a _custom_ test runner,
+  [`run_android_tests.dart`](bin/run_android_tests.dart).
+
+### Scope of Testing
+
+The tests in this package are end-to-end tests that _specifically_ exercise the
+engine and Android-specific embedding code. They are not unit tests for the
+engine or the framework, and are not designed to test the behavior of the
+framework or specific plugins, but rather to simulate the use of a framework
+or plugins downstream of the engine.
+
+In other words, we test "does the engine work on Android?" without a dependency
+on either the Flutter framework, Flutter tooling, or any specific plugins.
+
+See also:
+
+- [Unit tests for the Android embedder](../../../shell/platform/android).
+
+### Golden Comparisons
+
+Many of the Android-specific interactions with the engine are visual, such as
+[external textures](https://api.flutter.dev/flutter/widgets/Texture-class.html)
+or [platform views](https://docs.flutter.dev/platform-integration/android/platform-views),
+and as such, the tests in this package use golden screenshot file comparisons to
+verify the correctness of the engine's output.
+
+For example, in [`ExternalTextureTests_testMediaSurface`](https://flutter-engine-gold.skia.org/search?corpus=flutter-engine&include_ignored=false&left_filter=name%3DExternalTextureTests_testMediaSurface&max_rgba=0&min_rgba=0&negative=true&not_at_head=false&positive=true&reference_image_required=false&right_filter=&sort=descending&untriaged=true), a [video](app/src/main/assets/sample.mp4) is converted to an external texture and displayed in a Flutter app. The test takes a screenshot of the app and compares it to a golden file:
+
+<img 
+  alt="Two pictures, the top one Flutter and the bottom Android"
+  src="https://github.com/flutter/flutter/assets/168174/e2c34b88-d03d-4732-87e4-a86c97d006c5"
+  width="300"
+/>
+
+_The top picture is the Flutter app, and the bottom picture is just Android._
```

Comment:
It would be nice to have a link to https://github.com/flutter/engine/tree/main/tools/compare_goldens here, maybe a "see also" note.

Reply:
Done!